zeroeq/http/server.cpp

```diff
@@ -176,6 +176,25 @@ class Server::Impl : public detail::Sender
             uri.setHost(hostname);
         }
 
+        if (!servus::Servus::isAvailable())
+            return;
+
+        _service.set(KEY_INSTANCE, detail::Sender::getUUID().getString());
+        _service.set(KEY_USER, getUserName());
+        _service.set(KEY_APPLICATION, detail::getApplicationName());
+        _service.set("Type", "ZeroEQ");
+        if (!_session.empty())
+            _service.set(KEY_SESSION, _session);
+
+        const servus::Servus::Result& result =
+            _service.announce(uri.getPort(), getAddress());
+
+        if (!result)
+        {
+            ZEROEQTHROW(std::runtime_error("Zeroconf announce failed: " +
+                                           result.getString()));
+        }
+
         if (uri_.getHost() != uri.getHost() || uri_.getPort() == 0)
```

Comment:
It is, unless Zeroconf announcement fails in which case the ctor throws anyways.